controllers/v1/library/observationController.js

```diff
@@ -0,0 +1,133 @@
+    * @class
+*/
+module.exports = class Observation {
+    
+    constructor() {}
+
+    static get name() {
+        return "Observation";
+    }
+
+    /**
+    * @api {get} /assessment/api/v1/library/observation/list List of observation solutions
+    * @apiVersion 1.0.0
+    * @apiName List of observation solutions
+    * @apiGroup Observation Solutions Library
+    * @apiSampleRequest /assessment/api/v1/library/observation/list
+    * @apiUse successBody
+    * @apiUse errorBody
+    * @apiParamExample {json} Response:
+    * {
+    * "message" : "Successfully fetched observations solutions list",
+    * "status": 200,
+    * "result": {
+    * "data": [
+    * {
+    * "_id": "5ed5ec4dd2afa80d0f616460",
+    * "externalId": "AFRICA-ME-TEST-FRAMEWORK-TEMPLATE-BASE-SOLUTION",
+    * "name": "AFRICA-ME-TEST-FRAMEWORK",
+    * "description": "AFRICA-ME-TEST-FRAMEWORK"
+    * }
+    * ],
+    * "count": 1
+    * }
+    * }
+
+    */
+
+      /**
+      * List of observation solutions
+      * @method
+      * @name list
+      * @param {Object} req - All requested Data.
+      * @returns {JSON} returns a list of templates observation solution.
+     */
+
+    async list(req) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                
+                let observationSolution = 
+                await observationHelper.list( 
+                  req.searchText, 
+                  req.pageSize, 
+                  req.pageNo,  
+                );
+
+                return resolve(observationSolution);
+            } catch (error) {
+                return reject({
+                    status: error.status || httpStatusCode.internal_server_error.status,
+                    message: error.message || httpStatusCode.internal_server_error.message,
+                    errorObject: error
+                });
+            }
+        })
+    }
+
+    /**
+    * @api {get} /assessment/api/v1/library/observation/details/:librarySolutionId Details of observation solution.
+    * @apiVersion 1.0.0
+    * @apiName Details of observation solution
+    * @apiGroup Observation Solutions Library
+    * @apiSampleRequest /assessment/api/v1/library/observation/details/5ed5ec4dd2afa80d0f616460
+    * @apiUse successBody
+    * @apiUse errorBody
+    * @apiParamExample {json} Response:
+    * {
+    * "message": "Successfully fetched observation solution details",
+    * "status": 200,
+    * "result": {
+    * "creator": "Aman Karki",
+    * "about": "AFRICA-ME-TEST-FRAMEWORK",
+    * "questions": [
+    * "Does the library register show books being issued to at least 50% students in last 2 months?",
+    * "Does the school have a school development plan?",
+    * "Are the targets in the form of specific improvements and data-driven?",
+    * "Is there a documentation of the professional development opportunities the school leaders have engaged in?",
+    * "Is there a documented teacher professional learning plan?"
+    * ]
```

Comment:
@aman-tunerlabs - Will the linkTitle and linkUrl be in the response?

Reply:
@akash-above  For observation solution we dont require linkTitle and linkUrl.

Reply:
@aman-tunerlabs - Can you help me point to any place where they have said that explicitly they don't want it for observations? Or also have we put any condition like only solutions with type assessment can have these values in the backend while creating a solution?

Reply:
@akash-above  You can check the story video and we also discuss at the first meeting.

Reply:
@aman-tunerlabs - OK. Let's still have it, just in case they ask for it in future we don't need to make any changes. If its there in a solution we should pass it on to frontend and inform the same to Rejneesh as well.